### DISCUSSION TITLE
Troubleshooting 'over temperature error' with HSR-C robot during delicate mode grip operation

### DISCUSSION THREAD

**Discussion by KaitoNakatsuji** [HumanSupportRobot/QandA#78] [Q&A] [answered]
I'm experiencing an 'over temperature error' with the HSR-C robot and am seeking troubleshooting advice.
I am performing a grip using the "delicate mode" with the HSR-C.
`self._gripper.apply_force(req.value, delicate=True)`
However, the robot remains stationary when gripping, and I encounter an "over temperature error".
This issue occurs about 50 to 70% of the time.

I've reviewed the FAQ and tried the suggested solutions, but my issue remains unresolved.
[FAQ](https://docs.hsr.io/hsr_faq/hsr_common_faq.html)

> -- Question
> "Motor over temperature.(ExxxError/CoreTemp)"と頭部ディスプレイに表示される。
> -- Answer
> カバーを開けて、対象の軸に異物が挟まって動きにくくなっていないか確認してください。

Is this an issue specific to the HSR-C?
I can run the same program on my HSR-B without any problems, so I don't believe it's an issue with the source code.

Information regarding HSR-C

- HSR Serial Number 51
- /etc/opt/tmc/robot/version

> RELEASE_VERSION=22.04.01
> ROBOT_VERSION=HSRC-PHASE1
> LOCALE=ja_JP.UTF-8
> BUILD_DATE=20220511_000000_+0000

I would greatly appreciate any guidance or assistance you can provide on this matter. Thank you in advance for your time and help.

![Screenshot from 2023-08-31 17-55-24](https://github.com/HumanSupportRobot/QandA/assets/104056835/b72a7bc9-fdb4-453c-af2b-da1ce68fb6b6)

**Comment by yutaro-ishida** [comment]
At first, I would like to know which joint raises temperature error.
Could you show me the screen shot of `Diagnostics Viewer` in `rqt` when the error is occuring.

**Reply by KaitoNakatsuji** [reply]
Thank you for your reply. I have attached a screenshot regarding the joint temperature error. I identified an issue with the hand_motor_joint and subsequently checked the hand motor according to the FAQ. If you need any additional information, please let me know.

![HSR-C_rqt_Diagnostics_Viewer](https://github.com/HumanSupportRobot/QandA/assets/104056835/e39e7e89-2ac6-4d8e-8178-75ed78445211)

**Reply by yutaro-ishida** [reply]
Thanks. Yes, the error is definitely at the `hand_motor_joint`.

I would like to try your code on my HSR-C. 
Could you please share the code for the applicable part?

**Reply by KaitoNakatsuji** [reply]
The command I am using is as follows:
`gripper.apply_force(0.79, delicate=True)`
I've set the first argument to 0.79 because it switches to a power grasp mode when the value is 0.8 or higher.
One example of a problem occurs when trying to grip a sphere with a radius of about 3 cm.
If the object is within a size that can be grasped, the same issue is likely to occur.

**Comment by k-takeshita** [comment, marked as answer]
It's possible that the deterioration of the hand cover may have prevented proper force estimation.
Is it possible to remove the hand cover, perform calibration while it's removed, and attempt grasping?

**Reply by KaitoNakatsuji** [reply]
Thank you for your advice.
I have executed the calibration and confirmed the improvement. After the calibration, the grasping was successful in 10 out of 10 attempts. 
[Here](https://docs.hsr.io/hsrc_user_manual/howto/calibration/hand_force_calibration.html#id1) are the steps.
